Identity/Identity.API/Controllers/IdentityController.cs

```diff
@@ -0,0 +1,143 @@
+using Microsoft.Net.Http.Headers;
+using System.Security.Claims;
+
+namespace Recipes.API.Controllers;
+
+[ApiController]
+public class IdentityController(IdentityDatabaseContext dbContext) : ControllerBase
+{
+    [HttpGet]
+    [Route("users")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<UserEntity>>> GetAllRecipes()
+    {
+        IEnumerable<UserEntity> users;
+        try
+        {
+            users = await dbContext.Users.ToListAsync();
+        }
+        catch
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+
+        return Ok(users);
+    }
+
+    [HttpGet]
+    [Route("user/{id}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<UserEntity>>> GetAllRecipes(int id)
+    {
+        User user;
+
+        try
+        {
+            UserEntity? userEntity = await dbContext.Users.FindAsync(id);
+
+            if (userEntity is null)
+                return NotFound();
+
+            user = new User
+            { 
+                Username = userEntity.Username,
+                Email = userEntity.Email
+            };
+        }
+        catch
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+
+        return Ok(user);
+    }
+
+    [Route("register")]
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<UserEntity>>> RegisterUser(UserCreateRequest userCreateRequest)
+    {
+        UserEntity userToCreate = new UserEntity()
+        {
+            Username = userCreateRequest.Username,
+            PasswordHash = HashingHelper.HashPassword(userCreateRequest.Password),
+            Email = userCreateRequest.Email,
+            Roles = UserRole.Member | UserRole.Admin,
+        };
+
+        try
+        {
+            await dbContext.Users.AddAsync(userToCreate);
+            await dbContext.SaveChangesAsync();
+        }
+        catch
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+
+        return Ok(JWTHelper.GenerateJwtToken(userToCreate));
+    }
+
+    [HttpPost]
+    [Route("login")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<UserEntity>>> LoginUser(UserLoginRequest userLoginRequest)
+    {
+        string jwt;
+
+        try
+        {
+            UserEntity? user;
+            user = await dbContext.Users.Where(u => u.Username.Equals(userLoginRequest.Username)).FirstOrDefaultAsync();
```

Comment:
```suggestion
            var user = await dbContext.Users.Where(u => u.Username.Equals(userLoginRequest.Username)).FirstOrDefaultAsync();
```

Nitpick, but you not necessarily need to define this variable separately.

Reply:
Compressed this.